compass/Coordinator.java

```diff
@@ -333,37 +341,90 @@ private void createAndBroadcastMilestone(String trunk, String branch) throws Arg
    */
   private boolean validateTransactionsToApprove(String trunk, String branch) {
     if (validatorAPIs.size() > 0) {
-      boolean isConsistent = validatorAPIs.parallelStream().allMatch(api -> {
+
+      return validatorAPIs.parallelStream().allMatch(api1 -> {
         CheckConsistencyResponse response;
         try {
-          response = api.checkConsistency(trunk, branch);
+          response = getCheckConsistencyResponseWithRetires(trunk, branch, api1);
           if (!response.getState()) {
-            log.error("{} reported invalid consistency: {}", api.getHost(), response.getInfo());
+            log.error("{} reported invalid consistency: {}", api1.getHost(), response.getInfo());
           }
           return response.getState();
-        } catch (ArgumentException e) {
+        } catch (InterruptedException e) {
           e.printStackTrace();
           return false;
         }
       });
-
-      return isConsistent;
     }
 
     //nothing was checked so validation can't fail
     return true;
   }
 
+  private GetNodeInfoResponse getNodeInfoWithRetries() throws InterruptedException {
```

Comment:
Instead of all these functions, can we create a general `performAPICallWithRetries` that receives the API to call as a parameter? We avoid code duplication and we would improve readability.

Reply:
the problem is typing. this can be implemented in Jota quite simply in the future.
@GalRogozinski do you have a suggestion for this?